catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/RoleRepository.java

```diff
@@ -180,24 +186,16 @@ public void storeRelationships(Role role) {
   }
 
   public ResultList<Role> getDefaultRolesResultList(UriInfo uriInfo, Fields fields)
-      throws GeneralSecurityException, UnsupportedEncodingException {
+      throws GeneralSecurityException, IOException {
     List<Role> roles = getDefaultRoles(uriInfo, fields);
     return new ResultList<>(roles, null, null, roles.size());
   }
 
-  private List<Role> getDefaultRoles(UriInfo uriInfo, Fields fields) {
-    List<Role> roles =
-        daoCollection.roleDAO().getDefaultRoles().stream()
-            .map(
-                json -> {
-                  try {
-                    return withHref(uriInfo, setFields(JsonUtils.readValue(json, Role.class), fields));
-                  } catch (IOException e) {
-                    LOG.warn("Could not parse Role from json {}", json);
-                  }
-                  return null;
-                })
-            .collect(Collectors.toList());
+  private List<Role> getDefaultRoles(UriInfo uriInfo, Fields fields) throws IOException {
+    List<Role> roles = new ArrayList<>();
+    for (String roleJson : daoCollection.roleDAO().getDefaultRoles()) {
+      roles.add(withHref(uriInfo, setFields(JsonUtils.readValue(roleJson, Role.class), fields)));
+    }
     if (roles.size() > 1) {
```

Comment:
can't we avoid this ever happening when the admin tries to set the default role?

Reply:
Technically, this PR with DB Transaction while swapping default role prevents getting into a situation where we have multiple default roles.

My testing pattern has been the following:
- Start up OM with DataConsumer as default role
- Set another role R as default role and observe that DataConsumer is no longer default and user relationships have been updated. 
- Set DataConsumer as default role and obseve that R is no longer default and user relationships have been updated.

I did a few iterations of these locally with PostMan and ended up converting the logic into a test (RoleResourceTest.defaultRole)


I left the log line as is for detection purpose, in case there are any unforeseeable edge cases being missed out (as the code evolves).